buildSrc/src/main/java/ConcatenateJniHeaders.java

```diff
@@ -0,0 +1,55 @@
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Concatenates JNI headers generated during Java compilation into a single <pre>native.h</pre> file,
+ * for easier consumption by the native code.
+ */
+public abstract class ConcatenateJniHeaders extends DefaultTask {
+
+    /**
+     * The directory containing the generated JNI headers.
+     *
+     * Contains one header file per class with a native method.
+     */
+    @InputDirectory
+    public abstract DirectoryProperty getJniHeaders();
```

Comment:
This should really be a file collection, as the task does not really care where the files are or how they are arranged. It just needs a set of files.

Reply:
It is a `ConfigurableFileCollection` now.